File: samples/client/3_0_3_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/NumberTypeMatchesNumbersTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.schemas.validation.MapUtils;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class NumberTypeMatchesNumbersTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(JsonSchemaKeywordFlags.onlyFormat());

    @Test
    public void testAFloatIsANumberPasses() throws ValidationException {
        // a float is a number
        final var schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance();
        schema.validate(
            1.1d,
            configuration
        );
    }

    @Test
    public void testAnIntegerIsANumberPasses() throws ValidationException {
        // an integer is a number
        final var schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance();
        schema.validate(
            1,
            configuration
        );
    }

    @Test
    public void testAStringIsStillNotANumberEvenIfItLooksLikeOneFails() {
        // a string is still not a number, even if it looks like one
        final var schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance();
        try {
            schema.validate(
                "1",
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testABooleanIsNotANumberFails() {
        // a boolean is not a number
        final var schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance();
        try {
            schema.validate(
                true,
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testAFloatWithZeroFractionalPartIsANumberAndAnIntegerPasses() throws ValidationException {
        // a float with zero fractional part is a number (and an integer)
        final var schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance();
        schema.validate(
            1.0d,
            configuration
        );
    }

    @Test
    public void testNullIsNotANumberFails() {
        // null is not a number
        final var schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance();
        try {
            schema.validate(
                (Void) null,
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testAStringIsNotANumberFails() {
        // a string is not a number
        final var schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance();
        try {
            schema.validate(
                "foo",
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testAnArrayIsNotANumberFails() {
        // an array is not a number
        final var schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance();
        try {
            schema.validate(
                Arrays.asList(
                ),
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testAnObjectIsNotANumberFails() {
        // an object is not a number
        final var schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance();
        try {
            schema.validate(
                MapUtils.makeMap(
                ),
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }
}

File: samples/client/3_0_3_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/NumberTypeMatchesNumbersTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeyword
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class NumberTypeMatchesNumbersTest {
    companion object {
        val configuration = SchemaConfiguration(disabledKeywordFlags=setOf(JsonSchemaKeyword.FORMAT))
    }

    @Test
    fun testAFloatIsANumberPasses() {
        // a float is a number
        val schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance()
        schema.validate(
            1.1,
            configuration
        )
    }

    @Test
    fun testAnIntegerIsANumberPasses() {
        // an integer is a number
        val schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance()
        schema.validate(
            1,
            configuration
        )
    }

    @Test
    fun testAStringIsStillNotANumberEvenIfItLooksLikeOneFails() {
        // a string is still not a number, even if it looks like one
        val schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    "1",
                    configuration
                )
            }
        )
    }

    @Test
    fun testABooleanIsNotANumberFails() {
        // a boolean is not a number
        val schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    true,
                    configuration
                )
            }
        )
    }

    @Test
    fun testAFloatWithZeroFractionalPartIsANumberAndAnIntegerPasses() {
        // a float with zero fractional part is a number (and an integer)
        val schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance()
        schema.validate(
            1.0,
            configuration
        )
    }

    @Test
    fun testNullIsNotANumberFails() {
        // null is not a number
        val schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    null,
                    configuration
                )
            }
        )
    }

    @Test
    fun testAStringIsNotANumberFails() {
        // a string is not a number
        val schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    "foo",
                    configuration
                )
            }
        )
    }

    @Test
    fun testAnArrayIsNotANumberFails() {
        // an array is not a number
        val schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    listOf<Any?>(
                    ),
                    configuration
                )
            }
        )
    }

    @Test
    fun testAnObjectIsNotANumberFails() {
        // an object is not a number
        val schema = NumberTypeMatchesNumbers.NumberTypeMatchesNumbers1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    mapOf<Any?, Any?>(
                    ),
                    configuration
                )
            }
        )
    }
}
